fire/cli/niv/_regn.py

```diff
@@ -256,42 +285,57 @@ def gama_beregning(
     varliste = doc["gama-local-adjustment"]["coordinates"]["cov-mat"]["flt"]
     varianser = [float(var) for var in varliste]
     assert len(koter) == len(varianser), "Mismatch mellem antal koter og varianser"
+    tg = gyldighedstidspunkt(projektnavn)
+    return (punkter, koter, varianser, tg)
 
-    # Vi overskriver midlertidigt "Fasthold"-søjlen nedenfor, så vi tager en
-    # kopi og retablerer søjlen før resultatreturnering
-    fastholdsøjle = arbejdssæt["Fasthold"].copy()
 
-    # Skriv resultaterne til arbejdssættet
-    arbejdssæt["uuid"] = ""
-    arbejdssæt["Udelad publikation"] = ""
-    arbejdssæt["Fasthold"] = "x"
-    arbejdssæt["Ny kote"] = None
-    arbejdssæt["Ny σ"] = None
-    arbejdssæt["Δ-kote [mm]"] = None
-    arbejdssæt["Opløft [mm/år]"] = float("NaN")
-    arbejdssæt["System"] = "DVR90"
-    tg = gyldighedstidspunkt(projektnavn)
-    arbejdssæt = arbejdssæt.set_index("Punkt")
+# ------------------------------------------------------------------------------
+def gama_beregning(
+    punkter: list[str],
+    koter: list[float],
+    varianser: list[float],
+    arbejdssæt: List[float],
+    n_punkter: int,
+    tg: Timestamp,
+) -> List:
+
+    arbejdssæt = np.array(arbejdssæt)
```

Comment:
Det her med at håndtere dataelementer som nummererede søjler er ikke supergodt for kodens læselighed: Et NumPy-array er fornuftigt når alle elementer er ensfortolkede (altså som elementer i en matrix), men som du benytter det her mister vi overblikket over meningen med det enkelte dataelement.

Det er ikke det helt store problem her nede i funktionen selv, hvor de variabelnavne der skrives ind i arrayet giver tilstrækkelig hinting til at koden er læselig, men oppe i det kaldende program er den returnerede identitetsinformation reduceret til en stak indekstal, som ikke giver nogen information  om hvad det er man piller ud af den pågældende søjle i arrayet. **Der bliver det meget nemt at komme til at trække information fra en forkert søjle, hvilket så er meget vanskeligt at fejlfinde.**

I *ad hoc* tilfælde er løsningen at returnere en `list[dict]`, hvor hvert element af listen er en dict med sigende navne for hvert dataelement:

```python
arbejdselement = {'ny_kote': ny_kote, 'delta': Delta, 's.d.': sqrt(var), ...}
arbejdssæt.append(arbejdselement)
```

Men i mere *færdigbagte* løsninger er en `list[dataclass]`, med en til lejligheden nybagt dataclass, at foretrække, så man kan snakke om `arbejdselement.delta`, frem for `arbejdssæt[i, 7]`, eller `arbejdselement["delta"]`.

Her befinder vi os nok på kanten mellem *ad hoc* og *færdigbagt*, så om det er det ene eller det andet, der er løsningen, må være en smagssag. Men hvis du kigger op i den kaldende kode, hvor de returnerede data pakkes ud og anvendes, og forestiller dig hvad der skal til hvis vi en dag ændrer lidt på datamodellen, så tror jeg du vil kunne se at den af dine efterfølgere, der kommer til at sidde med den opgave, vil takke dig for at du droppede "magiske søjlenumre".